File: abc391a/abc391a.hs
{-# LANGUAGE OverloadedStrings #-}

main :: IO ()
main = solve <$> getLine >>= putStrLn

solve :: String -> String
solve = map opDir

opDir :: Char -> Char
opDir 'N' = 'S'
opDir 'S' = 'N'
opDir 'E' = 'W'
opDir 'W' = 'E'
opDir _ = ' '

File: abc391a/abc391a.ml
open Printf
open Scanf

let solve d =
  let f c = if c = 'N' then 'S'
            else if c = 'S'then 'N'
            else if c = 'E'then 'W'
            else 'E' in
  String.map f d

let () =
  scanf "%s " solve |> printf "%s\n"
